source/core/TDCodecVia.cpp

```diff
@@ -273,7 +274,9 @@ TypeRef TDViaParser::ParseType(TypeRef patternType)
     }
     
     Boolean bTypeFunction = _string.ComparePrefix('(');
-    if ((tt == TokenTraits_SymbolName) && (!bTypeFunction)) {
+    if (typeFunction.ComparePrefixCStr(tsEnumTypeToken) && (typeFunction.Length()==4 || isdigit(typeFunction.Begin()[4]))) {
```

Comment:
Having magic numbers like 4, makes the code more difficult to read. Use constants in this case, with a self explanatory name, that way we avoid the need of adding a comment.